apps/site/pages/en/learn/typescript/introduction.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Introduction to TypeScript
+layout: learn
+authors: sbielenica, ovflowd, vaishnav-mk, AugustinMauroy
+---
+
+# Introduction to TypeScript
+
+## What is TypeScript
+
+**[TypeScript](https://www.typescriptlang.org)** is an open-source language maintained and developed by Microsoft. It's loved and used by a lot of software developers around the world.
+
+Basically, it's a superset of JavaScript that adds new capabilities to the language. The most notable addition is static type definitions, something that is not present in plain JavaScript. Thanks to types, it's possible, for example, to declare what kind of arguments we are expecting and what is returned exactly in our functions or what's the exact shape of the object that we are creating. TypeScript is a really powerful tool and opens a new world of possibilities in JavaScript projects. It makes our code more secure and robust by preventing many bugs before the code is even shipped - it catches problems during code development and integrates wonderfully with code editors like Visual Studio Code.
+
+We can talk about other TypeScript benefits later, let's see some examples now!
+
+## First TypeScript code
+
+Take a look at this code snippet and then we can unpack it together:
+
+<!--
+  Maintainers note: this code is duplicated in the next article, please keep them in sync
+-->
+
+```ts
+type User = {
+  name: string;
+  age: number;
+};
+
+function isAdult(user: User): boolean {
+  return user.age >= 18;
+}
+
+const justine: User = {
+  name: 'Justine',
+  age: 23,
+};
+
+const isJustineAnAdult: boolean = isAdult(justine);
```

Comment:
I think a more realistic use of TS would not define types here, and instead take advantage of type inferring.
```suggestion
const justine = {
  name: 'Justine',
  age: 23,
} satisfies User;

const isJustineAnAdult = isAdult(justine);
```